code/modules/cm_marines/orbital_cannon.dm

```diff
@@ -562,8 +569,26 @@ var/obj/structure/ship_rail_gun/almayer_rail_gun
 	rail_gun_ammo.detonate_on(target)
 	cannon_busy = FALSE
 
-/obj/structure/ship_rail_gun/ex_act()
+/obj/machinery/ship_rail_gun/ex_act()
 	return
 
-/obj/structure/ship_rail_gun/bullet_act()
+/obj/machinery/ship_rail_gun/bullet_act()
 	return
+
+/obj/machinery/ship_rail_gun/proc/fire_laser_battery(turf/T, mob/user)
+	set waitfor = 0
+	if(cannon_busy)
+		return
+	if(ORBITAL_LASER_COST > machine_current_charge)
+		to_chat(user, "<span class='warning'>[src]'s APC has run out of charge.</span>")
```

Comment:
`machine_current_charge` is the `obj` internal battery, not the APC cell.

Reply:
Was from an earlier implementation; forgot to change it